component/prometheus/operator/common/component.go

```diff
@@ -106,18 +106,27 @@ func (c *Component) Update(args component.Arguments) error {
 	return nil
 }
 
+// NotifyClusterChange implements component.ClusterComponent.
+func (c *Component) NotifyClusterChange() {
+	c.mut.Lock()
```

Comment:
nit: is there any benefit to changing this to a RW mutex and using RLock/RUnlock?